### ISSUE TITLE
ENHANCEMENT: Set a ""default" mount_point

### ISSUE THREAD

**Issue by johnnybubonic** [hvac/hvac#674]
When one is using an instance of `hvac.Client` primarily with multiple operations on a single/specific mountpoint, it can become verbose/tedious to specify this same `mount_point` multiple times. (Think multiple operations of listing and updating secrets across multiple paths within the same mount, for instance.)

Would it be a lot of work to have an attribute on the instance that specifies the default `mount_point`? The way I see this potentially being the most user-friendly and realistic is being able to instantiate an appropriate `hvac.api.VaultApiBase` subclass (such as `hvac.api.secrets_engines.kv_v2.KvV2`) with a `__init__(self, mount_point = None)`. If mount_point is None, it'd use the `DEFAULT_MOUNT_POINT` value in each respective subclass (with 'mount_point' in each method still being able to override this and falling back to the class' attribute instead of the file-global `DEFAULT_MOUNT_POINT`).

Thoughts? Am I describing this clearly? I feel like I'm not, heh.

**Comment by jeffwecan** [comment]
Yeah that makes plenty of sense and I imagine wouldn't be too terribly difficult to implement. 👍🏻

**Comment by johnnybubonic** [comment]
(The second (`None` default per-method) or third (hybrid) designs above would retain backwards compatibility.)

**Comment by eytanhanig** [comment]
Any progress on this? It's a real quality of life improvement.

**Comment by Stogas** [comment]
Agree, this is a very useful enhancement.

**Comment by mddataminr** [comment]
Paying customer here - This would be great to have!

**Comment by yaroslav-nakonechnikov** [comment]
so cool, 4 years passed. and still waiting for it :)

**Comment by PanDas3** [comment]
still waiting :)

```
import hvac
client = hvac.Client(
    url='http://localhost:63740',
    token='token')

client.secrets.kv.v2.configure(mount_point='infra')
secret_version_response = client.secrets.kv.v2.read_secret_version(
    path='authentik',
)

print(secret_version_response)
```

and
`client.secrets.kv.v2.configure(mount_point='infra')`

 that not working

`hvac.exceptions.InvalidPath: no handler for route "secret/data/authentik". route entry not found., on get http://localhost:63740/v1/secret/data/authentik`

but as a previous friend said, when i'll change 
C:\Users\ \<User>\AppData\Roaming\Python\Python314\site-packages\hvac\api\secrets_engines\kv_v2.py 

`DEFAULT_MOUNT_POINT = "secret"` 
for 
`DEFAULT_MOUNT_POINT = "infra"`

then it works.

I have no idea how it's possible that everyone survived for so many years 😄 
All people used default name "secret" ?

**Comment by PanDas3** [comment]
oh... now I know.

in every method u must use **mount_point**
```
engine_path = "infra"
secret_version_response = client.secrets.kv.v2.read_secret_version(
    path='authentik',
    mount_point=engine_path
)
```
